configs/stable_diffusion/README.md

```diff
@@ -82,6 +82,122 @@ image = StableDiffuser.infer(prompt)['samples'][0]
+- `sx`, `sy` **(int, int)**: The stride for computing dst sets. A higher stride means you can merge more tokens, **default setting of (2, 2) works well in most cases**. `sx` and `sy` do not need to divide image size.
+- `use_rand` **(bool)**: Whether or not to allow random perturbations when computing dst sets. By default: True, but if you're having weird artifacts you can try turning this off.
+- `merge_attn` **(bool)**: Whether or not to merge tokens for attention **(recommended)**.
+- `merge_crossattn` **(bool)**: Whether or not to merge tokens for cross attention **(not recommended)**.
+- `merge_mlp` **(bool)**: Whether or not to merge tokens for the mlp layers **(particular not recommended)**.
+
+For more details about the **tomesd** setting, please refer to [Token Merging for Stable Diffusion](https://arxiv.org/abs/2303.17604).
+
+Then following the code below, you can evaluate the speed-up performance on stable diffusion models or stable-diffusion-based models ([DreamBooth](../dreambooth/README.md), [ControlNet](../controlnet/README.md)).
+
+```python
+import time
+import numpy as np
+
+from mmengine import MODELS, Config
+from mmengine.registry import init_default_scope
+
+init_default_scope('mmagic')
+
+_device = 3
+work_dir = '/path/to/your/work_dir'
+config = 'configs/stable_diffusion/stable-diffusion_ddim_denoisingunet.py'
+config = Config.fromfile(config).copy()
+# # change the 'pretrained_model_path' if you have downloaded the weights manually
+# config.model.unet.from_pretrained = '/path/to/your/stable-diffusion-v1-5'
+# config.model.vae.from_pretrained = '/path/to/your/stable-diffusion-v1-5'
+
+# w/o tomesd
+config.model.tomesd_cfg = None
+StableDiffuser = MODELS.build(config.model).to(f'cuda:{_device}')
+prompt = 'A mecha robot in a favela in expressionist style'
+
+# inference time evaluation params
+size = 512
+ratios = [0.75, 0.75]
+samples_perprompt = 5
+
+t = time.time()
+for i in range(100//samples_perprompt):
+    image = StableDiffuser.infer(prompt, height=size, width=size, num_images_per_prompt=samples_perprompt)['samples'][0]
+    if i == 0:
+        image.save(f"{work_dir}/wo_tomesd.png")
+print(f"Generating 100 images with {samples_perprompt} images per prompt, without ToMe speed-up, time used : {time.time() - t}s")
+
+for ratio in ratios:
+    # w/ tomesd
+    config.model.tomesd_cfg = dict(ratio=ratio)
+    sd_model = MODELS.build(config.model).to(f'cuda:{_device}')
+
+    t = time.time()
+    for i in range(100//samples_perprompt):
+        image = sd_model.infer(prompt, height=size, width=size, num_images_per_prompt=samples_perprompt)['samples'][0]
+        if i == 0:
+            image.save(f"{work_dir}/w_tomesd_ratio_{ratio}.png")
+
+    print(f"Generating 100 images with {samples_perprompt} images per prompt, merging ratio {ratio}, time used : {time.time() - t}s")
+```
+
+Here are some performance comparisons:
```

Comment:
How about including information about the GPU device that was used to evaluate the performance?